File: C GCC 5.4.1/ABC/ABC133/ABC133_A_v01.c
// [contest]  ABC 133
// [task]     A
// [URL]      https://atcoder.jp/contests/abc133/tasks/abc133_a
// [compiler] C (GCC 5.4.1)
// [status]   https://atcoder.jp/contests/abc133/submissions/6316771 : AC

#include <stdio.h>

int min (a, b)
{
	if (a > b) { return b; }
	else       { return a; }
}

int calculate_cost (int num_traveler, int cost_train, int cost_taxi)
{
	return min(num_traveler * cost_train, cost_taxi);
}

int main (void)
{
	// variables for main process
	int N, A, B;
	int rtvl_scanf;

	// STEP.01
	// read out the given data
	rtvl_scanf = scanf("%d %d %d", &N, &A, &B);

	// STEP.02
	// output the answer of this task
	printf("%d\n", calculate_cost(N, A, B));

	// STEP.END
	return 0;
}

/* EOF */

File: Haskell/ABC/ABC133/ABC133_A_v01.hs
-- [contest]  ABC 133
-- [task]     A
-- [URL]      https://atcoder.jp/contests/abc133/tasks/abc133_a
-- [compiler] Haskell (GHC 7.10.3)
-- [status]   https://atcoder.jp/contests/abc133/submissions/6538151 : AC

-- modules to import --

import qualified Data.ByteString.Char8
import qualified Data.Maybe
import           Prelude

-- functions for this task --

readInteger'    = Prelude.fst . Data.Maybe.fromJust . Data.ByteString.Char8.readInteger
readListInteger = Prelude.map readInteger' . Data.ByteString.Char8.words

getInteger     = readInteger'    <$> Data.ByteString.Char8.getLine
getListInteger = readListInteger <$> Data.ByteString.Char8.getLine

task_A :: Integer -> Integer -> Integer -> Integer
task_A numPeople costTrain costTaxi
        | costTrainAll < costTaxi = costTrainAll
        | Prelude.otherwise       = costTaxi
        where
                costTrainAll = costTrain * numPeople

-- the main process is as follows --

main :: IO ()
main = do

        -- STEP.01
        -- read out the given data
        [n, a, b] <- getListInteger

        -- STEP.02
        -- calculate & output the answer of this task
        Prelude.print $ task_A (n) (a) (b) 

-- EOF --
